app/src/main/java/org/mozilla/fenix/home/HomeFragmentStore.kt

```diff
@@ -125,7 +127,12 @@ private fun homeFragmentStateReducer(
             tip = action.tip,
             recentBookmarks = action.recentBookmarks,
             recentTabs = action.recentTabs,
-            historyMetadata = action.historyMetadata
+            historyMetadata = if (action.historyMetadata.isNotEmpty() && action.recentTabs.isNotEmpty()) {
+                val recentSearchGroup = action.recentTabs.find { it is SearchGroup } as SearchGroup?
+                action.historyMetadata.filterNot { it.title.equals(recentSearchGroup?.searchTerm, true) }
```

Comment:
Maybe we could have this in function as it looks pretty similar to what we are doing on line 158:166